sunpy/io/tests/test_fits.py

```diff
@@ -95,3 +98,26 @@ def test_extra_comment_write(tmpdir):
     outfile = tmpdir / "test.fits"
     sunpy.io.fits.write(str(outfile), data, header)
     assert outfile.exists()
+
+
+def test_map_fits_write_with_comments(tmpdir):
+    data, header = sunpy.io.fits.read(AIA_171_IMAGE)[0]
+    header['COMMENT'] = 'De-rotation: ON; Un-distortion: ON; Re-centering: ON; Re-sizing: OFF; co' \
+                        'rrection for cosmic-ray hits; correction front/side intensity implemente' \
+                        'd for mod L; RSUNerr=5.0 pixels; dpath=/home/jsoc/cvs/Development/JSOC/p' \
+                        'roj/lev1.5_hmi/apps/; linearity=1 with coefficients updated on 2014/01/1' \
+                        '5; smooth=1; propagate eclipse bit from level 1; use of larger crop radi' \
+                        'us look-up tables'
+
+    temp_file = tmpdir / "temp.fits"
+    sunpy.io.fits.write(str(temp_file), data, header)
+
+    fits_file = astropy.io.fits.open(str(temp_file))
+    data, header = fits_file[0].data, fits_file[0].header
+
+    aiamap = sunpy.map.Map(data, header)
+
+    out_file = tmpdir / "test.fits"
+    aiamap.save(str(out_file))
+
+    assert out_file.exists()
```

Comment:
I think we can simplify this test. 

I would make the comment shorter, I don't think we need 6 lines of strings. 
We can remove the map import. The bug is with the `io.fits` and not `Map`.

You want this test to do two things in my mind. 

1. Able to save out `COMMENT` in the header
2. Check that when you read the file again, the header `COMMENT` is the same.

Reply:
The making comment shorter is fine.

But I am not able to recreate the issue by just calling sunpy.io.fits.write method. Somehow the issue does not come up if one directly saves it avoiding maps.

I am still working on it, may be a day or two :)

Reply:
There is no rush. 

If the issue is with Map.save, then the test should be `https://github.com/sunpy/sunpy/blob/master/sunpy/map/tests/test_mapbase.py`. Sorry about that. 

But `Map.save` calls `io.write_file(filepath, self.data, self.meta, filetype=filetype, **kwargs)`, so the issue should crop up using that directly.